build-system/babel-plugins/babel-plugin-transform-inline-isenumvalue/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Takes enum check calls (isEnumValue, assertEnumValue) and replaces them with
+ * a smaller form that may not require the enum object at all, or only its
+ * values without keys.
+ */
+
+module.exports = function (babel) {
+  const {types: t, template} = babel;
+
+  function maybeTransformAssertEnumValue(path) {
+    const callee = path.get('callee');
+    const object = callee.get('object');
+    const property = callee.get('property');
+    if (
+      !callee.isMemberExpression() ||
+      !object.isCallExpression() ||
+      !property.isIdentifier({name: 'assertEnumValue'})
+    ) {
+      return false;
+    }
+    const objectCallee = object.get('callee');
+    const assertFnName = `${objectCallee.node.name}Assert`;
+    const [enumId, subject, enumNameNode] = path.node.arguments;
+    if (assertFnName === 'devAssert') {
+      path.replaceWith(t.cloneNode(subject));
+      return true;
+    }
+    const {specifiers} = path.scope.getBinding(objectCallee.node.name).path
+      .parent;
+    let specifier = specifiers.find(
+      ({imported}) =>
+        imported.type === 'Identifier' && imported.name === assertFnName
+    );
+    if (!specifier) {
+      specifier = t.importSpecifier(
+        t.identifier(assertFnName),
+        t.identifier(assertFnName)
+      );
+      specifiers.push(specifier);
+    }
+    const enumName = enumNameNode?.value || enumNameNode?.name || 'enum';
+    path.replaceWith(
+      template.expression`
+        (
+          ${t.identifier(specifier.local.name)}(
+            isEnumValue(${t.cloneNode(enumId)}, ${t.cloneNode(subject)}),
+            ${t.stringLiteral(`Unknown ${enumName} value: "`)} +
+              ${t.cloneNode(subject)} + '"'
+          ),
+          ${t.cloneNode(subject)}
+        )
+      `()
+    );
+    return true;
+  }
+
+  return {
+    name: 'transform-inline-isenumvalue',
+    visitor: {
+      CallExpression(path) {
+        // Replace x().assertEnumValue() calls with xAssert(isEnumValue(), ...).
+        // The replacement is handled on a subsequent pass.
+        if (maybeTransformAssertEnumValue(path)) {
```

Comment:
Nit: we're already moving everything over to `xAssert` instead of `x().assert`. Maybe just convert the holdouts and remove `assertEnumValue`

Reply:
Good idea. Removed `assertEnumValue` across the board.